File: Restaurant Billing System/Restaurant_Billing_System.c
#include<stdio.h>
#include<string.h>
int main()
{
    const char valid_username[]="admin";
    const char valid_password[]="hi@123";
    char username[50],password[50];
    int login_attempts=0;
    printf("Welcome to Hi Restaurant.\n");
    printf("Please login to continue...... \n");
    while (login_attempts < 3)
    {
        printf("Enter username: ");
        scanf("%s",&username);
        printf("Enter password: ");
        scanf("%s",&password);
        if (strcmp(username, valid_username) == 0 && strcmp(password, valid_password) == 0)
        {
            printf("Login successful! \nAcces granted.\n");
            break;
        }
        else
        {
            printf("Invalid username or password. Try again.\n");
            login_attempts++;
        }
        if (login_attempts == 3)
        {
            printf("Too many login attempts! Exiting the program.\n");
            return 1;
        }
    }
    int meal_plan, order_no, qty, add_more;
    int total=0;
    float cgst, sgst, total_bill;

    char *breakfast_menu[]={"Idly","Dosa","Woda","Bonda","Puri","Tea/Coffee"};
    int breakfast_prices[]={50,60,70,60,70,30};

    char *lunch_menu[]={"Veg Thali","Non-Veg Thali","Chicken biryani","Mutton Biryani","Veg Biryani","Salad/Papad","Cold Drink"};
    int lunch_prices[]={280,150,230,300,250,45,90};

    char *dinner_menu[]={"Chicken Biryani","Mutton Biryani","Chicken Curry","Paneer Curry","Roti/Chapati","Soup"};
    int dinner_prices[]={230,300,300,250,20,120};

    char **menu;
    int *prices;
    int menu_size;
    int order_quantity[10]={0};
    printf("\nChoose your meal plan: \n");
    printf("1. Breakfast\n2. Lunch\n3. Dinner\n");
    if (scanf("%d", &meal_plan) != 1)
    {
        printf("Invalid input! Please enter a number.\n");
        while(getchar() != '\n');
        return 1;
    }

    switch (meal_plan)
    {
        case 1: 
            menu=breakfast_menu;
            prices=breakfast_prices;
            menu_size=sizeof(breakfast_prices)/sizeof(breakfast_prices[0]);
            break;
        case 2:
            menu=lunch_menu;
            prices=lunch_prices;
            menu_size=sizeof(lunch_prices)/sizeof(lunch_prices[0]);
            break;
        case 3:
            menu=dinner_menu;
            prices=dinner_prices;
            menu_size=sizeof(dinner_prices)/sizeof(dinner_prices[0]);
            break;
    }
    printf("\nMenu:\n");
    for (int i=0;i<menu_size;i++)
    {
        printf("%d. %s - %d/-\n",i+1,menu[i],prices[i]);
    }
    do
    {
        printf("\nEnter item no to order: ");
        if (scanf("%d", &order_no) != 1)
        {
            printf("Invalid input! Please enter a number.\n");
            while(getchar() != '\n');
            return 1;
        }
        if (order_no < 1 || order_no > menu_size)
        {
            printf("Invalid choice! Please try again.\n");
            continue;
        }
        printf("Enter quantity: ");
        if (scanf("%d", &qty) != 1)
        {
            printf("Invalid input! Please enter a number.\n");
            while(getchar() != '\n');
            return 1;
        }
        order_quantity[order_no-1] += qty;
        total += prices[order_no-1]*qty;
        printf("\nAdd more to your order? (1 for yes/ 0 for no): ");
        if (scanf("%d", &add_more) != 1)
        {
            printf("Invalid input! Please enter a number.\n");
            while(getchar() != '\n');
            return 1;
        }
    }while(add_more == 1);

    printf("\nGenerating your BILL...\n\n");
    printf("BILL Generated\n");
    printf("========================================\n");
    printf("Item                Quantity   Price\n");
    printf("----------------------------------------\n");
    for (int i=0;i<menu_size;i++)
    {
        if (order_quantity[i] > 0)
        {
            printf("%-18s %3d   %5d/-\n",menu[i],order_quantity[i],prices[i] * order_quantity[i]);
        }
    }
    printf("----------------------------------------\n");
    printf("Sum total                    : %d/-\n", total);

    cgst=total*0.05;
    sgst=total*0.05;
    printf("CGST 5%%                      : %.2f/-\n", cgst);
    printf("SGST 5%%                      : %.2f/-\n", sgst);
    total_bill=total+cgst+sgst;
    printf("Total Bill                   : %.2f/-\n", total_bill);
    printf("========================================\n");
    return 0;
}

File: Restaurant Billing System/Restaurant_Billing_System.py
valid_username='admin'
valid_password="hi123"
print("Welcome to Hi Restaurant.")
for i in range(3):
    username=input("Enter username: ")
    password=input("Enter password: ")
    if username == valid_username and password == valid_password:
        print("Login successful!\nAccess granted.")
        break
    else:
        print("Invalid username or password.\nPlease try again.")
        if i == 2:
            print("Too many wrong attempts. \nAcess denied.")
            exit()
breakfast_menu={
    "Idly": 50,
    "Dosa": 60,
    "Woda": 70,
    "Bonda":60,
    "Puri": 70,
    "Tea/Coffee": 30
}
lunch_menu={
    "Veg Thali": 280,
    "Non-Veg Thali": 150,
    "Chocken Biryani": 230,
    "Mutton Biryani": 300,
    "Veg Biryani": 250,
    "Salad/Papad":45,
    "Cold Drink": 90
}
dinner_menu={
    "Chicken Biryani": 230,
    "Mutton Biryani": 300,
    "Chicken Curry": 300,
    "Panner Curry": 250,
    "Roti/Chapati": 20,
    "soup": 120
}
order_quantity,i =[],1
print("Enter your meal plan: ")
print("1. Breakfast\n2. Lunch\n3. Dinner")
while True:
    choice=input("Enter your choice (1/2/3): ")
    if not choice.isdigit():
        print("Invalid input! Please enter a number.")
        continue
    meal_plan=int(choice)
    if meal_plan == 1:
        menu=breakfast_menu
        print("\nBreakfast Menu:")
        break
    elif meal_plan == 2:
        menu=lunch_menu
        print("\nLunch menu: ")
        break
    elif meal_plan == 3:
        menu=dinner_menu
        print("\nDinner menu: ")
        break
    else:
        print("Invalid choice! Please enter (1/2/3). for meal plan.")
menu_items=list(menu.items())
menu_size=len(menu_items)
orders=[]
while True:
    print("/nAvailable Menu:")
    for idx, (item, price) in enumerate(menu_items, start=1):
        print(f"{idx}. {item} - Rs. {price}/-")
    order_input=input("Enter item number to order: ")
    if not order_input.isdigit():
        print("Invalid input! Please enter a number.")
        continue
    order_no=int(order_input)
    if order_no < 1 or order_no > menu_size:
        print("Invalid item number!")
        continue
    qty_input=input(f"Enter the quantity for {menu_items[order_no-1][0]}: ")
    if not qty_input.isdigit():
        print("Invalid input! Please enter a number.")
        continue
    qty=int(qty_input)
    orders.append((menu_items[order_no-1][0], qty, menu_items[order_no-1][1]))
    while True:
        more=input("Do you want to order more items? (y/n): ").lower()
        if more == 'y':
            break
        elif more == 'n':
            break
        else:
            print("Invalid input! Please enter only 'y or 'n'.")
    if more == 'n':
        break
import time
print("Generating your BILL........")
time.sleep(2)
print("BILL generated.")
print("="*40)
print(f"{'Item':<20}{'Quantity':<10}{'Price'}")
print("-"*40)
for item, qty, price in orders:
    print(f"{item:<20}{qty:<10}{qty*price}/-")
    print("-"*40)
subtotal=sum(qty*price for _, qty, price in orders)
cgst=round(subtotal*0.05,2)
sgst=round(subtotal*0.05, 2)
total=round(subtotal+cgst+sgst,2)
print(f"{'Sum Total':<30}: {subtotal}/-")
print(f"{'CGST':<30}: {cgst}/-")
print(f"{'SGST':<30}: {sgst}/-")
print(f"{'Total Bill':<30}: {total}/-")
print("="*40)